src/modules/populateData.js

```diff
@@ -0,0 +1,238 @@
+        headerHandler(commentDiv, commentHeader);
+        commentForm.reset();
+      });
+    });
+  });
+
+  document.querySelectorAll('.btn-reservation').forEach((el) => {
+    el.addEventListener('click', async () => {
+      const reservs = await fetchReservations(el.id);
+      reservationSection.classList.remove('hide-reservation');
+      const reservsGenerator = `<div class="reservation-container">
+        <button type="button" class="btn reservation-close-btn" id="close-reservation">
+          <i class="fas fa-times"></i>
+        </button>
+        <div class="reservation-img">
+          <img
+            src="${reservs.image.original}"
+            alt="${reservs.name}"
+          />
+        </div>
+        <div class="reservation-description">
+          <h3>${reservs.name}</h3>
+          <div class="reservations">
+            <p>Genres: ${reservs.genres}</p>
+            <p>Language: ${reservs.language}</p>
+            <p>Premiered: ${reservs.premiered}</p>
+            <p>Country: ${reservs.network.country.name}</p>
+          </div>
+        </div>
+        <div class="popup-reservations">
+          <h3>Reservations<span id="counter-reservations"></span></h3>
+          <div class="list-of-reservations">
+          </div>
+        </div>
+        <form method="post" class="add-reservation">
+          <h3>Add a reservation</h3>
+          <div class="input-field">
+          <input type="text" placeholder="Your name" class="input" name="name" id="nameReservation"/>
+          </div>
+          <div class="input-field">
+          <label for="startDate">Start date:</label>
+          <input type="text" placeholder="yyyy-mm-dd" class="input" name="startDate" id="startDate"/>
+          </div>
+          <div class="input-field">
+          <label for="endDate">End date:</label>
+          <input type="text" placeholder="yyyy-mm-dd" class="input" name="endDate" id="endDate"/>
+          </div>
+          <div class="btn-container">
+            <button type="submit" class="btn" id="reserve">Reserve</button>
+            <p class="alarm-form-reservations"></p>
+          </div>
+        </form>
+      </div>`;
+      reservationSection.innerHTML = reservsGenerator;
+      const reserveSubmit = document.getElementById('reserve');
+      reserveSubmit.addEventListener('click', async (event) => {
+        const alarmFormReservations = document.querySelector('.alarm-form-reservations');
+        const nameReservation = document.getElementById('nameReservation').value;
+        const startReservation = document.getElementById('startDate').value;
+        const endReservation = document.getElementById('endDate').value;
+        if (nameReservation && startReservation && endReservation) {
+          alarmFormReservations.innerHTML = '';
+          const regEx = /^\d{4}-\d{2}-\d{2}$/;
+          if (startReservation.match(regEx) && endReservation.match(regEx)) {
+            alarmFormReservations.innerHTML = 'Reservation Completed';
```

Comment:
- [ OPTIONAL] Most times successful messages have the color to be green and errors have their colors to be red so I think it is better if you make the `Reservation Completed` alert her have green color to differentiate the itself from the error massages :blush: